### PR TITLE
Ignore 10000 speed upper limit in Custom Game

### PR DESCRIPTION
**Pull request by TheSupervillain**
https://www.smogon.com/forums/threads/change-custom-game-speed-mechanics.3686665/

### CODE REVIEW

File: sim/pokemon.ts [hunk 1]
```diff
@@ -562,7 +562,7 @@ export class Pokemon {
 			stat = this.battle.runEvent('Modify' + statTable[statName], this, null, null, stat);
 		}
 
-		if (statName === 'spe' && stat > 10000) stat = 10000;
+		if (statName === 'spe' && stat > 10000 && !this.battle.format.id.includes('customgame')) stat = 10000;
```
Comment (DaWoblefet) [inline sim/pokemon.ts:565]: Pretty sure this should be ``!this.debugMode``.

Comment (TheSupervillain) [reply sim/pokemon.ts:565]: Well, `!this.battle.debugMode`, but yeah I get your point.

**Comment by Zarel** [comment]
Good work!

But `debugMode` should not affect mechanics. Can you change this to detect a `trunc` override, instead?

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**